text/0000-localizing-api-messages.md

```diff
@@ -0,0 +1,164 @@
+* Message Format - Refers to the standard used for formatting the message values in the translation file
+
+### In Scope Requirements/Use cases
+- Return localized messages based on the value passed in the accept-language header
+- Handle static messages with placeholder(s)
+
+### Out of Scope Requirements/Use cases
+- Naming convention to be used for keys
+- Process for managing (decoupled from lokalise.com) translations
+- Returning formatted (HTML/Markdown) messages
+- Usage of [soft hyphen](https://wiki.folio.org/display/I18N/How+To+translate+FOLIO#HowTotranslateFOLIO-Softhyphentobreakwords) to break messages
+- Support customization per tenant
+- Process  for [back porting](https://wiki.folio.org/display/I18N/Backport) API/Backend messages similar to what we have for the front end 
+- Support for [controlled vocabulary](https://issues.folio.org/browse/UXPROD-3148) (runtime data/data coming from DB tables. E.g. Patron Groups for tenants)
+
+## Detailed Explanation/Design
+#### API Protocol
+* accept-language header value MUST be in ll_CC format. , where ll is a two-letter language code,
+  and CC is a two-letter country code.
+* A module MUST return messages in the language specified in the HTTP "Accept-language" request header
+  (see IETF RFC 7231 section 5.3.5).  If the requested language is not available, a module
+  MUST return the message in the en language.
+* When handling messages with placeholders, replace the placeholders with the actual value on the server side before
+    sending it back to the client
+* Reliance on the lang query string parameter is DEPRECATED and SHOULD be updated to use the "Accept-language" header
+  at the earliest possible development sprint.
+
+#### Translation files
+* Translation file format will depend on the language that is used to develop the module
+  * Java, Groovy - properties
+  * JavaScript, nodejs - json
+  * Any other language - Please contact Technical Council for further guidance
+* Translation files MUST be placed under translations/\<Backend Module Name\>, relative to the root of the repo.
+  The files are being stored under a folder named after the module/repo to avoid having conflicts when using the same
+  message key across multiple backend modules. 
+* Translation file name MUST be [language code]_[country code].[extension] 
+  where `language code` is an ISO 639 two-letter language code, `country code` is an ISO 3166 two-letter country code 
+  and `extension` is the appropriate extension for the file format. For example, in `ll-CC.json`, 
+  ‘ll’ is the language code, ‘CC’ country code and `json` is the extension
+#### Translations
+* Translation message keys MUST be a valid string 
+* Message Values in the translation file MUST be formatted according to the [ICU](https://icu.unicode.org)  standard
+#### Language/Runtime
+* Module SHOULD use language/framework specific recommended best practices for loading translation at runtime
+
+
+
+## Rationale and Alternatives
```

Comment:
I think the rationale (inc trade offs) for the chosen approach should be split from the other considered options (maybe they could be moved to an appendix).

It would make it more obvious to the reader which option was chosen (if it isn't apparent from the above) and that the trade offs had been considered. 

At the moment, the `Risks and Drawbacks` section is lost within these option and comes across to me like an after thought. 

I acknowledge that this is likely a side effect of this RFC being written after the decision was made. However I think if we also want this to be an exemplar of an RFC.

(I think these same challenges are apparent for the Kafka ADR as well, that is not currently a good example)